src/lightning/pytorch/loops/fit_loop.py

```diff
@@ -94,6 +95,7 @@ def __init__(
 
         self._data_source = _DataLoaderSource(None, "train_dataloader")
         self._combined_loader: Optional[CombinedLoader] = None
+        self._combined_loader_states: Optional[List[Dict[str, Any]]] = None
```

Comment:
Having this default to an empty list should be equivalent and avoid the typing issues around None checks